http/responseemitter.go

```diff
@@ -147,12 +160,28 @@ func (re *responseEmitter) Close() error {
 	return nil
 }
 
-func (re *responseEmitter) SetError(v interface{}, errType cmdkit.ErrorType) {
-	err := re.Emit(&cmdkit.Error{Message: fmt.Sprint(v), Code: errType})
-	if err != nil {
-		log.Debug("http.SetError err=", err)
-		panic(err)
+func (re *responseEmitter) CloseWithError(err error) error {
+	re.once.Do(func() { re.preamble(err) })
+
+	re.l.Lock()
+	defer re.l.Unlock()
+
+	e, ok := err.(*cmdkit.Error)
+	if !ok {
+		e = &cmdkit.Error{Message: err.Error()}
+	}
+
+	if re.streaming || e.Code == cmdkit.ErrFatal {
```

Comment:
Aren't all errors "fatal" now? That is, can't we just *always* use the trailer now?